### ISSUE TITLE
Taking over the alternate buffer with a numbered temp file

### ISSUE THREAD

**Issue by mike-hearn** [airblade/vim-gitgutter#238]
I tracked down a really strange error I was experiencing to this plugin.

If I have two or more files open, and I edit a file and then try and switch to the alternate buffer (using `<C-^`), a temp file has taken over the alternate buffer with a file named with in the pattern of `[number].[extension of last used file]`, and it opens that temp file instead of the actual alternate buffer.

For example, if the current buffer was `views.py`, and I attempted to switch to the alternate buffer, it would open `2.py`. If I switch back to `views.py` and then switch again, it will open `5.py`.

Does this sound familiar and/or like something that might have been introduced in a recent update?

**Comment by mike-hearn** [comment]
Here's a GIF screencast of the issue actually happening: http://i.imgur.com/esaCC4B.gif

This GIF is probably a little hard to follow, so here's what I'm doing:

1. Open my .vimrc file (just as an example file)
2. Open a second blank file
3. I demonstrate that I can switch between them without triggering the bug
4. I add a blank line to the top of the .vimrc file, so it's been edited
5. I start switching between the current buffer and the alternate buffer. In theory, this should just switch between the two open files
6. Instead, it opens files titled 7, 10, 13, 16, 19, etc.

This bug was recently introduced, because older versions of vim-gitgutter did not do this, but unfortunately I'm not sure which version it was. If necessary, I can do a deeper dive and try to reproduce this bug with a bunch of different version.

**Comment by airblade** [comment]
Thanks for the excellent bug report.

This problem was introduced in 0cb1e41 (see #232 and #233 for more information).  In a nutshell, during the realtime diff, instead of passing the buffer contents to `git-diff` on stdin we write it to a temp file and hand the file to `git-diff`.

As you spotted this had the unintended side-effect of changing the buffer's alternate buffer.  Fortunately it's an [easy fix](https://github.com/airblade/vim-gitgutter/commit/3420cbf6cb75c95e169a58d578c95e7da59a9b35#diff-6bae73666dcc383d0d5dd9d3cffd8741R33).
